src/client/pages/Payment/useRedirectURL.ts

```diff
@@ -0,0 +1,31 @@
+import { useLocation } from 'react-router'
+import { useMemo, useEffect } from 'react'
+
+const QUERY_PARAMETER = 'redirect_url'
+const SESSION_STORAGE_KEY = `_hedvig_${QUERY_PARAMETER}`
+
+export const useRedirectURL = () => {
+  const query = useQuery()
+
+  useEffect(() => {
+    const value = query.get(QUERY_PARAMETER)
+    if (value) {
+      window.sessionStorage.setItem(SESSION_STORAGE_KEY, value)
+    }
+  }, [query])
+
+  return useMemo(() => {
+    const value = query.get(QUERY_PARAMETER)
+    if (value) return value
+
+    const storeValue = window.sessionStorage.getItem(SESSION_STORAGE_KEY)
+    if (storeValue) return storeValue
+
+    return undefined
+  }, [query])
+}
+
+const useQuery = () => {
+  const { search } = useLocation()
+  return useMemo(() => new URLSearchParams(search), [search])
```

Comment:
Cost of `new URLSearchParams` is trivial, why not just create it and extract redirect URL on every render?

Reply:
haha just picked this from react route docs

Reply:
I have to wrap in memo though, otherwise it will cause useEffect to run over and over....

Reply:
Probably can be simplified like this
```
const search = useLocation()
const redirectUrl = // extract from search
useEffect(() => ..., [redirectUrl]
```
No memo needed since redirectUrl is a primitive